### ISSUE TITLE
Moderation shortage

### ISSUE THREAD

**Issue by joepie91** [nodejs/community-committee#33]
(Creating a new issue since it was indicated in #20 that anything pertaining to number of moderators would be considered off-topic there.)

I've noticed a shortage of moderation in #Node.js in the past few weeks, unfortunately. There have been a few cases where it was difficult to get hold of an active channel moderator, to deal with a problematic user - in fact, there's such a problematic user active in the channel *right now* that refuses to leave.

Is this a known issue?

**Comment by Trott** [comment]
I don't think Node.js Foundation ever took on #Node.js IRC channel officially. I think it's a separate, independent entity, although [managing it in a more official capacity has certainly been discussed](https://github.com/nodejs/TSC/issues/121#issuecomment-244561818).

That doesn't answer your question, but it hopefully provides some context and explains a difficulty.

cc'ing a few folks who are ops in #Node.js who are also sometimes active in the nodejs GitHub discussions: @ashleygwilliams @feross @isaacs @brycebaril

**Comment by joepie91** [comment]
It was indicated in nodejs/TSC#121 (specifically, in [this post](https://github.com/nodejs/TSC/issues/121#issuecomment-282501130)) that further discussion should be directed here - hence why I'm filing this issue here :)

I honestly have no idea where else I'd file this anyway.

**Comment by jasnell** [comment]
This is definitely something that needs to be figured out. There has been an active discussion around more "official" communication channels and the community committee is definitely the right place to solve those issues. @nodejs/community-committee, let's make sure this is on the agenda for the next call (if it's not already, I'm still catching up after being a bit disconnected the past couple of weeks)

**Comment by Fishrock123** [comment]
@isaacs has more or less handed over #Node.js to myself/@emilyrose.

If we want to add more people so that we can cover odd times better (or any other reason really) then let's do it. 
Most of the existing OPs are more inactive than active, fwiw.

**Comment by MylesBorins** [comment]
I'm perpetually idling and often by computer. Willing to help with op duties

**Comment by nebrius** [comment]
I'm removing the `cc-agenda` label because there are some broader structural changes that need to happen before we can really discuss this issue.

**Comment by jfhbrook** [comment]
I need to +1 this in a big way.

# From my perspective: For the last 2 years #node.js has had no active mods, full stop.

At some point, Max casually added @avianflu (I don't know the full timeline) and as we realized that the mods were asleep at the wheel, he started trying to step in despite the fact that he doesn't work in node anymore. Eventually, in the last 3 or 4 months, he asked me to help because I've been complaining about this problem non-stop on twitter pretty much every time a troll shows up.

Now, I'm *glad* to do this. Banning jerkwads is one of my primary skills! I'm way better at this than dealing with the minutiae of trying to run a community project through ~~github issues~~.

@Fishrock123 was nice enough to point me in the right direction(s) as far as access to repos and has been around for the assist recently thank goodness. But someone, somewhere, has been asleep at the wheel, and that's not OK. Even now, with 3 of us part-time, we're overwhelmed. Which isn't surprising, considering the room has ~1000 people in it.

Some things I would like to get cleared up:

1. Does the node.js CoC cover the irc room? Ts the irc room considered one of the spaces "owned" by the project?
2. Who used to moderate the node.js irc room? Are they existing or past members of the CTC?
3. Why have mods been afk for so long? If not part of the CTC, do they expect to be able to come back and assert authority after years of being absent? If part of the CTC, why have issues around moderation not been resolved?
4. If #node.js is a space controlled by the project, and presumably there are general purpose moderation resources, can we get a hand?

**Comment by rachelnicole** [comment]
@jfhbrook I use irccloud as a client, so while I'm always present in the room, I'm not always immediately there. Maybe we should redefine how reporting should be if someone violates something / needs moderation. Because while I might not be directly in front of my screen, I also have irccloud on my phone and can jump in at any moment.

**Comment by jfhbrook** [comment]
This is good! And something I would like to work on as well--my client is still irssi in a tmux session, and as you might imagine doesn't work great on phones. I can get decent coverage during the workday and most worknights, just by virtue of not leaving my apartment.

I don't expect perfect coverage, and in fact I was basically OOO the last 3 weeks due to medical issues. I just need help, is all.

Moderating IRC is challenging because it has to happen on 2 levels (and I'm making this up based on my experiences, if any of you here have some theory to throw behind this I'm definitely interested):

1. on-call: someone mentions you, and it sets off an alarm that makes your phone beep. You hop in and ban the spammer. This is the easy case, but one that has been the most difficult to deal with. Many of us only have visual bells. I can fix this, with some patience while I find the spoons to switch to irccloud.
2. participatory de-escalation: A lot of times, someone isn't actively spouting truly odious ideas but are instead getting in a fight with someone over a software opinion, or maybe someone is being just sexist enough to bother people, in which case you have to be actually paying attention in order to try and set the tone for the rest of the conversation, issue warnings, etc. Giving some of your more active users faith that you'll take care of something when pm'd ( @joepie91 is pretty good at this kind of thing) helps, but isn't a substitute for actually hanging out some odd Thursday night.

If you're volunteering to be involved in the irc room @rachelnicole that's fantastic! Same with @MylesBorins but I feel like a lot of people have volunteered and then haven't been around. But dunno, maybe I'm just being grouchy. Totally willing to start w/ a clean slate here.

**Comment by jfhbrook** [comment]
I want to ask about this for a second:

> @isaacs has more or less handed over #Node.js to myself/@emilyrose.

So am I to read this as:

1. The channel is not an official project space
2. The owners are officially (is this reflected in the access list at all?) @Fishrock123 and @Amorelandra 

If so, some immediate problems:

1. @baudehlo made some comments earlier today that indicated that, to me, at one point in time, @aredridel was basically running the show but that was all definitely a very long time ago and as far as I can tell there's been zero communication by anyone as to what the actual state is...maybe we should either (a) clean up the moderation list or (b) go through the effort to bring them into the fold? Possibly some combination thereof?
2. From the same conversation and others, it became clear to me that, if the irc room isn't a project space that the decision to adopt node's CoC as the official rules has not been reached by any kind of consensus (though with the amount of afk people on the moderation list, how could we?)

**Comment by rachelnicole** [comment]
@jfhbrook i agree with #1, we should re-review the moderation list and try and reach out to those who are lapsed. 

Also for #2, that's totally something that the Community Committee can vote on, for bringing it under the jurisdiction of the org. Perhaps someone else might have some more guidance?

**Comment by jfhbrook** [comment]
so idk freenode channel ownership rules (or even how its chanserv does that kinda thing, I've mostly worked with mibbit in terms of access lists), but it would probably be good to make sure we know who holds the keys in case someone feels like they're being usurped. With 20 mods I could see at least one of them going poorly.

**Comment by baudehlo** [comment]
I'm always around and always available at a ping in EST work hours. I don't actively read the channel because it's too active. I've been out for 2 weeks due to a new baby.

I'm not sure what the major concern is here and I disagree with @jfhbrook's shouty headline above about nobody being around for the last 2 years. I'm just a silent member of the channel 99% of the time.

**Comment by TimothyGu** [comment]
The lack of moderation on IRC channel **is a problem** and **is driving people away**. A member of the community has only recently described #node.js on another Freenode channel as "regularly explodes into flame wars" and eventually left because they "can't stand it anymore". IMO, IRC should indeed have a higher level of tolerance for language because of its instant nature, but some of the things that are said in the channel would have provoked a swift moderation response had it been on GitHub. (Just read the [logs](http://logs.nodejs.org/node.js/2017-08-31) and decide for yourselves.)

**Comment by jfhbrook** [comment]
@baudehlo congrats on the child!

but I haven't seen you take a single moderation action, even when we've mass pinged mods in the past. Even so, people have to be actively participating at least sometimes to be an effective mod, I'm sorry.

**Comment by baudehlo** [comment]
I have all my logs in irccloud going back years. When was this ping and who from?

If I'm at work when I'm pinged I respond, always.

Here's the full list of "pings" I have:

* I have one ping from kiraid on May 12th at 9:08pm. I'm not around at that time.
* Another from naezith on December 28th 2016 at 6:19pm. I wasn't around.
* I have sdegutis asking if I'm an op on August 28th at 2:06pm which I responded yes to, there were no further questions.
* I have a conversation with AvianFlu (another Op) about banning Avinash on September 20th, 2016.

That's literally all the PMs I have regarding ops on #Node.js, aside from questions from people unable to talk without registering.

The moderation actions I've taken have been light handed, telling people to cut it our or chill when required. Mostly I've been inactive because nobody has asked me to help out with a problem (and I don't actively read the channel because there's too much backlog to go through to keep up with everything).

As for driving people away, you need data to back that up rather than an anecdote. The reduction in content in #Node.js (I graphed it over the last 2 years to check) matches the general reduction in usage of IRC across multiple channels.

I'm not arguing that we don't need more moderators (we do, mostly to cover timezones), but if there's issues requiring moderation on #Node.js, nobody is contacting me about them, and I can prove it.

**Comment by joepie91** [comment]
@baudehlo I have pinged a number of ops (including you) in the channel several times in recent history:

```
[Saturday, April 08, 2017] [11:21:33 PM] <joepie91>	ping mscdex feross baudehlo
[Saturday, April 15, 2017] [09:49:31 PM] <joepie91>	feross: baudehlo: mscdex: around?
[Thursday, June 01, 2017] [08:36:42 PM] <joepie91>	ec: probably should. topic tends to lag behind :) cc mylesborins mscdex feross Fishrock123 baudehlo
[Thursday, June 01, 2017] [08:40:01 PM] <joepie91>	baudehlo: thanks :)
[Monday, July 10, 2017] [02:25:16 PM] <joepie91>	ping baudehlo, feross, mscdex, mylesborins - any of you happen to be around?
[Saturday, July 22, 2017] [06:27:14 PM] <joepie91>	paging baudehlo, feross, jfhbrook, mscdex, mylesborins, whichever op is currently awake.
```

Of these cases, you only responded on June 1, by updating the topic.

As for me personally, I typically only bother pinging ops in channel in extreme cases, because it virtually never results in an op response anyway (from either you or anybody else, leaving the reason for that aside), and there's a real risk of it further escalating the situation, as it has done in the past.

Pinging ops *privately* is something that I might more easily do... if it actually resulted in responses from the ops I pinged, and there was a clear indication of what ops are going to be available. Somebody never responding to pings in channel doesn't inspire confidence in me that they're going to respond to a ping in private, and it costs time and effort to reconcile/centralize things if multiple ops end up responding to a PM about the same incident.

On top of that, active coverage of the channel (ops responding to incidents of their own accord, without needing to be actively notified of incidents) is essentially nonexistent. All of this together means that there's a moderation problem, and that people stop bothering to report incidents. The massive access list, mostly containing inactive users, exacerbates that problem.

So far, the only op who *consistently* gets back to me about incidents, has been @AvianFlu - however, he's rarely around at the same time as I am, and often isn't around when incidents occur. That's not a point of criticism towards him personally (as he's in a different timezone), just an observation of how this makes rule enforcement difficult.

As far as I'm concerned, the following things need to happen:

1. Clean up the access list, removing everybody who has no intention of moderating the channel. This clarifies towards users who the contact points are for incidents.
2. Consistently respond to in-channel notifications as well as private notifications, ensuring that incidents have been handled.
3. Actively moderate the channel, keeping at least a cursory eye on what happens in the channel even if nobody is explicitly pinging ops.
4. As already said, enough ops to actually cover the channel, availability-wise. There don't need to be sub-minute response times, but having to wait 2 days for moderator resolution (if one occurs at all!) is just *too long*, for a channel of this size and importance.

**Comment by baudehlo** [comment]
#2 is hard because on a channel with lots of scrollback it's hard to find who said your name (maybe easier in other clients, I'm not sure). PM ops if you want them. I will respond, fast in EST business hours, slower in all other cases. I get PMs on my phone though, so you will likely have more luck that way.

Perhaps we should say that in the /topic too.

As I said, we do need more moderators, but finding them is HARD, and trying to call out the current ones on being inactive isn't a good approach.

**Comment by rachelnicole** [comment]
i think whatever we decide on it should definitely be reflected in a gist somewhere so people can see who to contact / about what, etc. 

i run irc cloud too, and often the only questions i get in pings are people asking me why other people banned them, or technical questions.

**Comment by jfhbrook** [comment]
I agree that @AvianFlu has been fantastic. I'm trying to be more generally available myself, but haven't been around as long, don't have my client set up yet, and have been dealing with some personal shit the last month or so--so bad timing.

I also agree that it would be nice to get some nite modz. We're pretty US-centric, and while ime traffic does decrease at night, there are still situations that happen at 3:00am eastern.

As previously mentioned, I feel pretty strongly that the best ops are going to be ones that are actively involved in the channel at least sometimes--I've deescalated a number of situations without anyone getting pinged, just by being Around. This is something that simply can't be done with an on-call model.

I have few concerns around alienating existing mods for calling them out on being afk, since worst case they're still afk. I suspect a number of them have written off irc and spend most of their time on curated slack spaces like wealljs, and just haven't bothered to deop themselves out of a sense of duty. If that's not the case--and it sounds like @baudehlo is overwhelmed by chat volume but does legitimately want to be available--then we can talk.

Also in agreement about updating messaging in our topic to clarify open issues. I think active mods have been a little shy with the topic, given the lack of a clear owner of the channel.

**Comment by jfhbrook** [comment]
Regarding an on-call, one idea, and I don't have much bandwidth for hacking these days, is a moderation bot that implements oncall schedules, so you'd `!oncall` and hilight mods that have registered as being available at those times.

**Comment by ljharb** [comment]
@jfhbrook just wanted to dispute your "2 years" comment; until last September or so, I've been a very active op in that channel, frequently enforcing the node CoC and moderating issues. During my 2 years or so being an op there were very few times I wasn't online when an issue occurred.

**Comment by jfhbrook** [comment]
I'll be the first to admit that my sense of time is very poor. I'll regularly confuse 6 weeks and 6 months in conversation, much to the frustration of my friends. In other words, I don't entirely dispute that.

**Comment by joepie91** [comment]
> @jfhbrook: Regarding an on-call, one idea, and I don't have much bandwidth for hacking these days, is a moderation bot that implements oncall schedules, so you'd !oncall and hilight mods that have registered as being available at those times.

You'd probably want `!ops`, then - that's the de facto standard for these kind of bot pings in most channels, and I've more than once seen a user attempt to use that command in #Node.js as well.

> @baudehlo: #2 is hard because on a channel with lots of scrollback it's hard to find who said your name (maybe easier in other clients, I'm not sure). PM ops if you want them.

I understand that this can be tricky, but it's not reasonable to expect random channel users (ie. not me) to PM 5-10 people on the access list in the hopes of catching one that's active. This is a problem that's easier to solve on the ops' side (client configuration that shows a channel highlight list or such), than on the users' side. Putting things in the topic isn't likely to help with that, either.

**Comment by jfhbrook** [comment]
I'm also going to throw out there, and I'm not necessarily advocating for anything, but @ljharb was at one point in time a mod, and while I don't remember how that went or why his @ got taken away (probably @Amorelandra can say more) I've seen him be pretty involved in the channel and having deescalated a number of situations. If we're going to be cleaning up the access list and adding people that are de facto more involved, this is something to consider.

Again, just throwing it out there.

**Comment by jfhbrook** [comment]
RE: !ops, is that smart enough to not ping me if I'm sleeping? Not critical for this kinda thing, jsut wondering.

**Comment by Amorelandra** [comment]
Hey everyone. I'm glad to see an increased interest in moderation of IRC.
It seems like we have an issue that might involve technical solutions as well as additional recruitment.

I'd also just like to mention that I'm almost always available to drop into IRC. I'm not currently idling a client in the channel, but I can be reached via twitter (my DMs are open).

**Comment by jfhbrook** [comment]
@Amorelandra something I would very much like to see is, from everyone on the access list, either a commitment to be on irc and participating in chat for at least a few hours a week, or voluntarily removing themselves from the access list. Considering that you're not even on freenode: how do you respond?

**Comment by Amorelandra** [comment]
@jfhbrook I'd respond by asking you to chill a little bit. I don't think it's reasonable to expect moderators to be online 24/7.

**Comment by jfhbrook** [comment]
No, but it's reasonable to expect moderators to be online and participating a couple times a week. It's possible we've just missed each other, but I have not ran into you on that channel in what feels like months. You're not the only one either.

**Comment by Amorelandra** [comment]
Yep, you've definitely been missing me, then.

I'm happy to chat through solutions, but the blame game doesn't seem like a productive endeavor.

**Comment by jfhbrook** [comment]
I don't want to blame people, I want mods that do more than idle, and I don't appreciate being told that the status quo is ok, because it's not, and I know this isn't just me--multiple people have already backed me up on this.

**Comment by Amorelandra** [comment]
> being told that the status quo is ok

who's saying this to you?

**Comment by jfhbrook** [comment]
All I know is that I'm getting resistance when I'm saying that mods are generally unavailable. I interpret this as being told that existing mod coverage is fine.

**Comment by Amorelandra** [comment]
I think we would all do better to ask clarifying questions rather than making assumptions.

What can I help clarify?

**Comment by joepie91** [comment]
> @jfhbrook: RE: !ops, is that smart enough to not ping me if I'm sleeping? Not critical for this kinda thing, just wondering. This is assuming that there's a canned solution we can deploy.

I'm not aware of any `!ops` bots that are availability-aware (ie. have an 'on call' system). It wouldn't be hard to write one, though.

**Comment by jfhbrook** [comment]
Sure:

1. @joepie91 has strong evidence that we've had extreme difficulty getting ahold of moderators at many hours of the day, and I can anecdotally confirm. Why is this? why shouldn't existing mods have to answer for this in some manner?
2. Out of people currently opped in the channel, I've seen @AvianFlu actively moderate, and myself participating casually at work and more in-depth at home and on weekends. I've hardly seen anyone else on the access list participating in the chat. I suspect many of them don't use IRC anymore. Why shouldn't we clean these people out of the access list, so it's more clear who can/will be able to take action?
3. I've said multiple times that being on-call isn't enough, since many issues get bad enough to turn off users but not so bad that they pull up the channel list and start pinging names. I seem to be getting pushback on this, and I don't understand why. Surely "DM people on twitter" isn't an generally acceptable means of accessing an irc mod.
4. Who actually owns the channel? Are they still involved in the management of the channel? If not, can we transfer ownership?
5. Why should I bother trying to help, if this is going to end up being a scenario where it's effectively just me and @AvianFlu ? This conversation is making me feel like I should throw in the towel on both moderation and participation in the irc room.

**Comment by jfhbrook** [comment]
@joepie91 starting to think that the on-call schedule component would be more trouble than it's worth. :) but worth talking through regardless.

**Comment by Amorelandra** [comment]
> Why is this? why shouldn't existing mods have to answer for this in some manner?

Because this isn't an inquisition, it's a call for solutions. People are allowed to have lives that exist outside of IRC. There's nothing to "answer for".

> Why shouldn't we clean these people out of the access list, so it's more clear who can/will be able to take action?

If the desire is for *more* moderation, why are you asking to remove access for existing moderators? There is no real limit to the number of people who can be on the list. It seems to make more sense to simply find additional folks willing to get involved.

> I seem to be getting pushback on this

What is giving you this perception?

> This conversation is making me feel like I should throw in the towel on both moderation and participation in the irc room.

I will admit that from my perspective you do seem rather aggressive with the way you're approaching the conversation. Perhaps it would be in your best interest to consider stepping away for the time being? It's rarely good to be involved in situations that actively cause you stress.

**Comment by joepie91** [comment]
> @Amorelandra Because this isn't an inquisition, it's a call for solutions. People are allowed to have lives that exist outside of IRC. There's nothing to "answer for".

Sure, they are, but when those lives become so time-consuming that people are unable to moderate the IRC channel, then perhaps they are not in a good position to be moderators for it?

> @Amorelandra: If the desire is for more moderation, why are you asking to remove access for existing moderators? There is no real limit to the number of people who can be on the list. It seems to make more sense to simply find additional folks willing to get involved.

I already answered this several comments ago: it is unclear who is available to deal with channel incidents, and people are discouraged from reporting issues when the people they message or ping consistently don't ever answer.

**Comment by Amorelandra** [comment]
> I already answered this several comments ago: it is unclear who is available to deal with channel incidents, and people are discouraged from reporting issues when the people they message or ping consistently don't ever answer.

This sounds like a problem in need of a technical solution. I like the idea of a way to ping all ops via a simple and well-understood command.

**Comment by ljharb** [comment]
I'm not sure tone policing is going to help keep the conversation productive.

**Comment by jfhbrook** [comment]
> Because this isn't an inquisition, it's a call for solutions. People are allowed to have lives that exist outside of IRC. There's nothing to "answer for".

Being an irc moderator is a *responsibility*. Either you **take on** that responsibility, or you don't. This doesn't fly.

> If the desire is for more moderation, why are you asking to remove access for existing moderators? 

It's extremely disheartening when those moderators don't actually show up to do anything. A name that doesn't lead to anything is more than useless, it's just noise. There needs to be some confidence level around thinking that pinging a mod will actually bring someone online. It's also unclear who I, as a mod, can talk to, when I need a second opinion.

Again, being a mod is a responsibility, not a badge that you get to wear so you feel cool.

> What is giving you this perception?

You, baudehlo, myles, even rachel, have all responded with "I'll respond to pings" as though that's enough. Few people have volunteered to participate, and yet want to keep their sherrif's badge.

> I will admit that from my perspective you do seem rather aggressive with the way you're approaching the conversation. Perhaps it would be in your best interest to consider stepping away for the time being? It's rarely good to be involved in situations that actively cause you stress.

If it seems like I'm being aggressive, it's because seeing a channel I love go down the shitter due to non-moderation is a problem that's been bothering me for a very long time, and it's only now that I've been able to find a venue to express this. But really, I don't appreciate the tone policing either. What I'm saying is valid, even if you don't like it.

>  I like the idea of a way to ping all ops via a simple and well-understood command.

Doing this manually, as people have done, rarely works. This is not in and of itself a solution, just a piece of the puzzle. That said, this is something I want to see, since in many clients pulling up the names list, searching for @s and pinging them one by one is hard work. Lower barriers of entry are always good.

**Comment by rachelnicole** [comment]
@Jfhbrook how is saying that I'll be responsive if someone pings me not participating? Legitimate question. I'm always online as part of my job as is a lot of other people on the list. Perhaps we just need to create a better call to action for situations that need to be moderated.

**Comment by jfhbrook** [comment]
@rachelnicole participation means hanging out in the irc room, answering questions, talking to people, noticing when prometh is being a jerk about angular again and reminding people that we all have different opinions about tech and that shitting on other peoples' work isn't really acceptable, etc.

You can't get that from idling.

**Comment by Amorelandra** [comment]
> Being an irc moderator is a responsibility. Either you take on that responsibility, or you don't. This doesn't fly.

What kind of answers are you looking for?

> sherrif's badge.

I'm not comfortable with this association.

> But really, I don't appreciate the tone policing either. What I'm saying is valid, even if you don't like it.

My apologies if it came across as "tone policing", but I'm trying to understand your perspective and ask for clarification. Nothing I have said has been an attempt to invalidate what you're saying. Please do not misinterpret my concern for your well-being as an attempt to police your tone.

**Comment by jfhbrook** [comment]
> my concern for your well-being

I don't believe for a second that this is in good faith.

> I'm not comfortable with this association.

Hey, I'm not a huge fan of the cops either. Insert whatever metaphor you prefer--the point is that the @ isn't something you get for being cool.

> What kind of answers are you looking for?

See my answer to rachel's question.

**Comment by rachelnicole** [comment]
@jfhbrook idk i think in this case, I disagree. I've been using irc for over half my life and have been in lots of channels that run like ours do. 
 
You don't need to be a moderator to tell people that people have different opinions on tech. Moderation is only necessary when users can't self moderate themselves.

I am willing to come up with a better solution than what exists though! It should be revisited

**Comment by Amorelandra** [comment]
> I don't believe for a second that this is in good faith.

OK, then! I'll be abstaining from further comment.

**Comment by joepie91** [comment]
> You don't need to be a moderator to tell people that people have different opinions on tech. Moderation is only necessary when users can't self moderate themselves.

Which occurs with some regularity, and there is no clear escalation path, nor is there active participation from channel operators to spot trouble *before* people walk out in frustration. Which is why this thread exists in the first place.

I'm not sure what is even being discussed now. It has been indicated repeatedly that there is a problem with the level of moderation in the channel, by multiple people, from multiple perspectives. It's clear that the current moderation policy is *not working*, so I don't get why this discussion seems to be devolving into what sounds a *lot* like "but it's fine how it is now".

Why are we still discussing the merits of the complaints, instead of finding concrete solutions to the problems that have already been pretty conclusively shown to exist? I don't feel like the complaints are taken seriously right now.

**Comment by baudehlo** [comment]
Joshua what you're missing from your "if you're not active then GTFO"
response is that not everyone is active all the time, and so taking that
attitude means you'll just kick someone off the list because they haven't
been active for N days/months, when they may have just been super busy at
work, or in their personal life, and will get back to being on #Node.js
soon. I've definitely been that way. I go through dips and peaks on it. So
you'd rather have me kicked off the list, and yet here I am back at it
today participating, so you bite off your nose to spite your face. Better
to just add more moderators than try and threaten the existing ones with
being kicked out if they don't participate.

If you're feeling a bit of moderator pushback, it's because this became a
bit of a witch hunt in which we had to defend ourselves. Nobody likes doing
that. It's not fun.

The !ops bot looks like a reasonable solution to me. Surely we can just add
a bot like that and try that as an interim solution until more people
volunteer to moderate?

**Comment by rachelnicole** [comment]
@joepie91 I just said "I am willing to come up with a better solution than what exists though! It should be revisited"

Should we close this issue and start a new one with actionable items / requests to keep better track of how to improve the policy?

**Comment by joepie91** [comment]
I don't see what closing this issue and opening a new one would accomplish. I've already provided a number of [actionable items](https://github.com/nodejs/community-committee/issues/33#issuecomment-326607286) for solving the moderation issues in the channel, based on over two years of active participation(!) in the channel, but instead people seem to be getting upset about the possibility of their `+o` being taken away.

I really don't get the fuss here. It's in the channel's best interest to keep the access list devoid of inactive users, as it's a standardized way to consult the users responsible for managing a channel (with or without an `!ops` bot), which means that having inactive entries on it will confuse and frustrate users, not to mention the security implications of keeping access around for people who don't need it from a pragmatic perspective.

Okay, sometimes life takes over, and sometimes that means one doesn't have time to spend on moderation for an extended period of time. What is the problem with removing access in the meantime? It can be re-added once somebody is available again. This is turning into a storm in a teacup, and by this point I can't help but feel that this is more about status symbols than about the channel's best interests.

Once again: this problem can be solved by cleaning up the access list, finding new operators who can commit to *participating actively in the channel* for some amount of time, and improving response consistency when ops are explicitly pinged. *These are the actionable items*, and as far as I'm concerned, anything that doesn't address these points is just noise. This thread has been going on for *several months* now with no concrete solutions in sight.

And yes, I am beginning to get frustrated, because this thread is devolving into what seems like interpersonal drama and non-constructively picking on specific words and phrases, when really it should be exclusively about how to best moderate the channel. I don't even really care *who* the ops are, so long as they can commit to the above, and act in the channel's best interests.

**Comment by rachelnicole** [comment]
Okay cool, I'm fine with stepping down from ops since the level of participation I'm willing to contribute seems to be not good enough for what you think is best for the channel.

Good luck.

**Comment by mcollina** [comment]
@rachelnicole please reconsider. IMHO this issue is about having _more_ ops rather than less ops.
I'm definitely not agreeing on the direction this conversation is taking.

Can we talk about _more_ ops, rather than a process to remove the existing ops?

**Comment by baudehlo** [comment]
I agree with Rachel, if that's the consensus then I'm happy to step down
too. Keeping adding/removing mods based on their activity level seems like
a daft idea to me, as you'll just remove people never to add them back. I
don't know why Sven is going all hyperbole on this. We've said we want more
moderators and better tools, but it seems to be the aim to get rid of the
current mod team and find a new one.

Good luck, sincerely.

**Comment by rachelnicole** [comment]
@mcollina i tried to steer the conversation into a different direction explicitly talking about improvements in a clean thread and it seemed to be not good enough. I think @joepie91 is under the impression that some people care about the 'status' of being an op, which I honestly couldn't care less about, I just volunteered last time someone asked if i would be willing to help.

**Comment by jfhbrook** [comment]
well hey, I'm out. It's clear that nobody in this group sees eye-to-eye with me when it comes to the commitment that being an op on irc should bring with it, and I'm unwilling to put in the effort I think is necessary if I get minimal backup.

**Comment by hackygolucky** [comment]
@jfhbrook Reading this scrollback, I see a lot of folks _agreeing_ with you on the need to fix a lot of problems you've generously spent your time identifying. The methods to which you were offering a solution they may have disagreed on, but we have to find some common ground here. 

Ultimately, this entire thing, a large OSS effort, is still almost entirely a volunteer effort. I am a paid employee of the Node.js Foundation, but there is _so_ much work to be done that much of my GitHub and IRC time is very much outside of any sort of work hours.  This project relies on people stepping up and offering the time they have available. The complication with moderators is that while _anyone_ can volunteer, it doesn't mean that (a) they have the skillset to do so and if they have the skillset, (b) they may not have a lot of time. Do we not allow folks to volunteer if they can't commit to enough hours? I think that was the heart of the conversation y'all were drilling down to, but I'd love to hear thoughts.

@joepie91 @rachelnicole @baudehlo @Amorelandra Thanks for offering up your time and ideas as well. I'd been added as a moderator but also have not gotten many pings. I've had some one on one conversations with folks to moderate, but I've also missed a few. I have felt the guilt around not being available enough, which is why [we had started this lovely thread](https://github.com/nodejs/community-committee/issues/33) AND [this one](https://github.com/nodejs/community-committee/issues/20).

**Comment by Amorelandra** [comment]
Thank you, @hackygolucky.

After talking through issues discussed in this thread I've done two things to improve my availability and address what I perceive as misgivings:
* I have switched to the use of IRCCloud so as to be alerted of pings via my mobile device as well.
* I have removed auto-op from my channel access flags, so that my name longer appears by default with the contentious "+o/sheriff's badge" that's been mentioned.

I'm also experimenting with a bot to make it much easier to alert available mods. No promises on this front, but interested parties should hit me up if they'd like to collaborate.

**Comment by jfhbrook** [comment]
@hackygolucky,

The only reason this channel has any traction at all is because of about a dozen people that spend almost all day in there, on a volunteer basis. Obvious examples include ljharb and joepie, but there are others. I get that volunteering takes time, but there are currently people taking that time even though they're largely powerless to fix problems when they see them.

Yes, I do think that moderating in irc should be a commitment, because moderating irc is difficult and when done poorly leads to channel quality going down the toilet. I've had to leave multiple irc channels over peoples' inability to moderate them or make them welcoming spaces. It's really sad to me that #node.js is basically on that list now.

As I've said multiple times, I don't expect people to be chatting 24/7, or not going on vacation, etc. But if someone is gone for months at a time and/or "doesn't really do irc anymore"? Yes, it would be appropriate to ask if they're being effective at the position that they volunteered for.

At any rate, I'm done, both with this thread and with the irc room. I'll be clicking the unfollow button.

**Comment by Amorelandra** [comment]
_[I'm happy to re-post this on another thread if appropriate]_

If you have any interest in helping to moderate the IRC channel, and you are at least somewhat active on a regular basis: let me know. You don't have to be an expert at IRC. I will be more than happy to show you the commands and explain our moderation strategy.

**Comment by baudehlo** [comment]
Why aren't ljharb and joepie mods?

**Comment by bnb** [comment]
@Amorelandra definitely interested in collaborating on that bot - happy to help however I can. 👍

**Comment by joepie91** [comment]
(I'm only around for a brief moment today - I'll write a more extensive post later, if people have further questions.)

@baudehlo: For me personally, the offer has been extended to me in the past, but I've turned it down due to conflict-of-interest issues; most notably, some of my [tutoring](http://cryto.net/~joepie91/code-review.html) customers come from #Node.js, and I don't want to create an impression that my services are in any way 'officially sanctioned'.

That having been said, the amount of customers that come in through #Node.js has dropped in recent times, in favour of people finding me through different venues. While I'm still a bit iffy about the possible conflicts of interest, I'd be willing to consider taking on +o if the needed moderation coverage can't be obtained otherwise.

Thoughts on this are welcome.

**Comment by Amorelandra** [comment]
@joepie91 I'm definitely still open to revisiting this offer if you feel it would be appropriate and we have support from other moderators.

**Comment by Trott** [comment]
It seems like perhaps this should be closed. Feel free to re-open (or leave a comment requesting that it be re-opened) if you disagree. I'm just tidying up and not acting on a super-strong opinion or anything like that.
